app/src/main/kotlin/com/wire/android/media/CallRinger.kt

```diff
@@ -28,16 +28,21 @@ class CallRinger @Inject constructor(private val context: Context) {
     }
 
     fun ring(resource: Int, isLooping: Boolean = true) {
+        mediaPlayer?.reset()
         mediaPlayer?.release()
         createMediaPlayer(resource, isLooping)
         mediaPlayer?.start()
     }
 
     fun stop() {
         try {
-            if (mediaPlayer?.isPlaying == true)
-                mediaPlayer?.stop()
-            mediaPlayer?.release()
+            mediaPlayer?.apply {
+                if (isPlaying) stop()
+                reset()
+                release()
+            }
+
+            mediaPlayer = null
         } catch (e: IllegalStateException) {
```

Comment:
What was the `exception` being thrown?

It would be cool to give more info about it on the PR description to share knowledge with the team too! I have no clue what was going on and how this fixes it 😄

Reply:
Added it to the PR description, but just in case:
`java.lang.IllegalStateException at android.media.MediaPlayer.isPlaying(Native Method)`